src/tables.jl

```diff
@@ -47,3 +47,585 @@ function schema_names(featuredefn::IFeatureDefnView)
+    geomnames = colnames[geomindices]
+
+    fieldindices = isa.(AGtypes, Tuple{OGRFieldType,OGRFieldSubType})
+    fieldtypes = AGtypes[fieldindices] # TODO consider to use a view
+    fieldnames = colnames[fieldindices]
+
+    # Create layer
+    layer = createlayer(name = name, geom = first(geomtypes))
+    # TODO: create setname! for IGeomFieldDefnView. Probably needs first to fix issue #215
+    # TODO: "Model and handle relationships between GDAL objects systematically"
+    GDAL.ogr_gfld_setname(
+        getgeomdefn(layerdefn(layer), 0).ptr,
+        first(geomnames),
+    )
+
+    # Create FeatureDefn
+    if length(geomtypes) >= 2
+        for (geomtype, geomname) in zip(geomtypes[2:end], geomnames[2:end])
+            creategeomdefn(geomname, geomtype) do geomfielddefn
+                return addgeomdefn!(layer, geomfielddefn) # TODO check if necessary/interesting to set approx=true
+            end
+        end
+    end
+    for (fieldname, (fieldtype, fieldsubtype)) in zip(fieldnames, fieldtypes)
+        createfielddefn(fieldname, fieldtype) do fielddefn
+            setsubtype!(fielddefn, fieldsubtype)
+            return addfielddefn!(layer, fielddefn)
+        end
+    end
+
+    return layer, geomindices, fieldindices
+end
+
+"""
+    _infergeometryorfieldtypes(sch, rows, spgeomcols, spfieldtypes)
+
+Infer ArchGDAL field and geometry types from schema, `rows`' values (for WKT/WKB cases) and `geomcols` and `fieldtypes` kwargs 
+
+"""
+function _infergeometryorfieldtypes(
+    sch::Tables.Schema{names,types},
+    rows,
+    spgeomcols::Union{Nothing,Vector{String},Vector{Int}},
+    spfieldtypes::Union{
+        Nothing,
+        Dict{Int,Union{OGRwkbGeometryType,Tuple{OGRFieldType,OGRFieldSubType}}},
+    },
+) where {names,types}
+    colnames = string.(sch.names)
+
+    # Convert column types to either geometry types or field types and subtypes
+    AGtypes =
+        Vector{Union{OGRwkbGeometryType,Tuple{OGRFieldType,OGRFieldSubType}}}(
+            undef,
+            length(Tables.columnnames(rows)),
+        )
+    for (j, (coltype, colname)) in enumerate(zip(sch.types, colnames))
+        if spfieldtypes !== nothing && j ∈ keys(spfieldtypes)
+            AGtypes[j] = spfieldtypes[j]
+        else
+            # we wrap the following in a try-catch block to surface the original column type (rather than clean/converted type) in the error message
+            AGtypes[j] = try
+                (_convert_cleantype_to_AGtype ∘ _convert_coltype_to_cleantype)(
+                    coltype,
+                )
+            catch e
+                if e isa MethodError
+                    error(
+                        "Cannot convert column \"$colname\" (type $coltype) to neither IGeometry{::OGRwkbGeometryType} or OGRFieldType and OGRFieldSubType",
+                    )
+                else
+                    throw(e)
+                end
+            end
+        end
+    end
```

Comment:
I tried, but the highlighted section is the only part of this function that I was able to comprehend. I.e. I'm afraid I found it too challenging to follow L172-282 of this file in this PR (at the time of this writing).

I can see the value of having a data-driven inference of geometries from WKT/WKB columns, but I cannot in good faith say that I'll know how to maintain it. For unblocking this PR, can we document it in a feature request for now, so the code will still be available if we want to go down this path?

Reply:
Well, I'm still around :-) And I could deactivate the WKT/WKB column detection by default via an `auto::Bool=false` kwarg which have it value `true` value mutually exclusive with `fieldtypes != nothing`

But here is an explanation, that I hope will clarify the code
#### 1. L145-L170: Parse stage 1 (based on schema) -> Identify column types for later layer population stage
#### 2. L172-L258: Parse stage 2 (based on data) for column that may be WKT or WKB columns and therefore identified as IGeometry columns for later layer population stage
```
For each column C of type string (which may be a WKT column) or binary (which may be a WKB column)
    1. Iterate though elements E of C until one E cannot be converted to IGeometry (except when = nothing or missing)
    2. If all element of C can be converted to IGeometry, mark C as a IGeometry column for later layer population stage
       Else leave it marked as a Field column 
```
You told me in a previous PR that you were not fond of ∈, ∉, ∩, ∪, ⊆, ⊈, ... I naturally use them since it . If it helps to maintain of the code, I can of course replace them by their plain function names.
#### 3. L260-L285 (induced by the `geomcols` kwarg addition): Enforce IGeometry columns to be as user specified and ensure that it is coherent with the actual data for later layer population stage. Raise an error otherwise, with more information than would be available at layer population stage (simple `convert MethodError`)
```
If there is a list of columns marked as IGeometry by the user (via `geomcols` kwarg)
    1. If there is one column in this list that has not been identified as an IGeometry column at parse stage 1 or 2 ERROR  
    2. For each column C identified as IGeometry as parse stage 1 or 2 and outside of this list
           If C can be converted to a Field column mark it so for later layer population stage
           Else ERROR
```
By the way, I have added another restriction on columns to be parsed at stage 2, by excluding columns specified in `fieldtypes` kwarg. See commit d08a47c

Reply:
I'll need more time to work through your code and comments -- thank you in advance for your initiative and patience!

Reply:
You have convinced me about the utility of the WKT/WKB parsing, and the lack of utility of `spfieldtypes`. But this PR is really hard to review for correctness.

I'll be able to approve this PR if we stash the WKT/WKB comments & code in an issue/follow-up PR.

Reply:
Done in commit e8560e4